soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -4620,6 +4641,69 @@ void CreateScrubMessages() {
     }
 }
 
+void CreateRupeeMessages() {
+    CustomMessageManager* customMessageManager = CustomMessageManager::Instance;
+    customMessageManager->AddCustomMessageTable(Randomizer::rupeeMessageTableID);
+    const std::vector<u8> rupees = { TEXT_BLUE_RUPEE, TEXT_RED_RUPEE, TEXT_PURPLE_RUPEE, TEXT_HUGE_RUPEE };
+    std::string rupeeText = " ";
+    for (u8 rupee : rupees) {
+        switch (rupee) {
+            case TEXT_BLUE_RUPEE:
+                rupeeText = "\x05\x03 5 #RUPEE#\x05\x00";
+                break;
+            case TEXT_RED_RUPEE:
+                rupeeText = "\x05\x01 20 #RUPEE#\x05\x00";
+                break;
+            case TEXT_PURPLE_RUPEE:
+                rupeeText = "\x05\x05 50 #RUPEE#\x05\x00";
+                break;
+            case TEXT_HUGE_RUPEE:
+                rupeeText = "\x05\x06 200 #RUPEE#\x05\x00";
+                break;
+        }
+        customMessageManager->CreateMessage(Randomizer::rupeeMessageTableID, rupee,
+            { TEXTBOX_TYPE_BLACK, TEXTBOX_POS_BOTTOM,
+                "You found" + rupeeText + " !",
+                "Du hast" + rupeeText + "  gefunden!",
+                "Vous obtenez" + rupeeText + " !"
+        });
+    }
+}
+
+std::string Randomizer::InsertRupeeName(std::string message, int language) {
```

Comment:
i'm not the biggest fan of the naming here

the message in the table isn't having the randomized rupee name inserted, we're creating a new string with the randomized rupee name in it

maybe just `GetRandomizedRupeeMessage`

Reply:
Hmm, I guess technically it does create a new string, but the only difference between the string it takes in and the string it returns is the rupee name. So it's not really "getting" a rupee message, because that's happening in GetRupeeMessage.

Same thing is happening with `FormatCustomMessage()` in CustomMessageManager.cpp. If we do want to rename this function to something like what Bria suggested, then we should probably also rename this one to `GetFormattedCustomMessage()` to stay consistent.

I'm not really sure what the best naming for this would be.

Reply:
I'm actually quite on board with 'RandomizeRupeeName'. And yes, they really are sometimes haha.